tests/integration/targets/openssh_cert/tests/key_idempotency.yml

```
@@ -63,13 +63,13 @@
         valid_from: always
         valid_to: forever
       register: second_signature_algorithm
-      when: openssh_version is version("8.7", "!=")
+      when: not (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] >= "9")
 
     - name: Assert second signature algorithm update causes change
       assert:
         that:
           - second_signature_algorithm is changed
-      when: openssh_version is version("8.7", "!=")
+      when: not (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] >= "9")
```

Otherwise `"10"` won't be `>= "9"` :-)
```suggestion
      when: not (ansible_facts['distribution'] == "RedHat" and (ansible_facts['distribution_major_version'] | int) >= 9)
```
Also how about adding the `when` to ` block` containing both tasks, and adding a comment that this `when` is needed for systems not allowing SHA1? (There could be more such systems in the future.)

---

Done